core/src/Cuda/Kokkos_Cuda_Parallel_Range.hpp

```diff
@@ -1026,7 +1061,8 @@ class ParallelScanWithTotal<FunctorType, Kokkos::RangePolicy<Traits...>,
         if (!m_result_ptr_device_accessible)
           DeepCopy<HostSpace, CudaSpace, Cuda>(
               m_policy.space(), m_result_ptr,
-              m_scratch_space + (grid_x - 1) * size / sizeof(int), size);
```

Comment:
I don't actually know where `int` came from in the original, but `sizeof(word_size_type)` is important here otherwise the result will be incorrect